receiver/awsfirehosereceiver/internal/unmarshaler/cwmetricstream/unmarshaler.go

```diff
@@ -36,49 +33,42 @@ func NewUnmarshaler(logger *zap.Logger) *Unmarshaler {
 	return &Unmarshaler{logger}
 }
 
-// Unmarshal deserializes the records into cWMetrics and uses the
-// resourceMetricsBuilder to group them into a single pmetric.Metrics.
-// Skips invalid cWMetrics received in the record and
-func (u Unmarshaler) Unmarshal(records [][]byte) (pmetric.Metrics, error) {
+// UnmarshalMetrics deserializes the records into CWMetric and uses the
+// ResourceMetricsBuilder to group them into a single pmetric.Metrics.
+// Skips invalid CWMetric received in the record.
+func (u Unmarshaler) UnmarshalMetrics(records [][]byte) (pmetric.Metrics, error) {
 	md := pmetric.NewMetrics()
-	builders := make(map[resourceAttributes]*resourceMetricsBuilder)
+	builders := make(map[ResourceAttributes]*ResourceMetricsBuilder)
 	for recordIndex, record := range records {
-		// Multiple metrics in each record separated by newline character
```

Comment:
Why did you remove this? Per https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-metric-streams-formats-json.html:

> In a CloudWatch metric stream that uses the JSON format, each Firehose record contains multiple JSON objects separated by a newline character (\n). Each object includes a single data point of a single metric.

Reply:
You are right, thanks for giving the reference. I couldn't find it before. I have fixed it now.